[EyuGongYi] iP

EyuGongYi opened the pull request:
# Diomon
> Why be basic when u can be cool - Someone cool

Diomon takes out all the excess bloatware, and frees your mind by keeping track of what you need to do. It's

- Lightweight
- text-based
- easy to learn
- made in Java
- completely ** FREE**!
- safe to use (Source: [Trust me](https://www.youtube.com/watch?v=dQw4w9WgXcQ))

All you need to do is,
1. Download it from here.
2. Double-click and run the it.
3. Add you tasks
4. Let it manage your task for you :innocent:

Features:
- [X] Managing Tasks
- [ ] Deleting Mac OS (Coming Soon)
- [ ] Valorant Hax (Coming Soon)

If you are a programmer, here is the main code of the program
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

EyuGongYi committed:
Add Diomon.java, Changed README.md

EyuGongYi committed:
Add echo feature

EyuGongYi committed:
Add ability to store and list text inputs

EyuGongYi committed:
Main.java: Refactored the echo logic. Add Task and TaskList

Echo logic needed to change to better fit the classes added.

Moved Greeting and bye(), Moved separation lines out

Makes logic consistent and modular.

EyuGongYi committed:
Add Task, TaskList, Commands, Deadline, Event, Todo.java

EyuGongYi committed:
Add delete command

EyuGongYi committed:
Add Automated testing

martout2002 commented on README.md:
````diff
@@ -1,24 +1,3 @@
-   | |_| | |_| |   <  __/
-   |____/ \__,_|_|\_\___|
-   ```
+This is a greenfield project by Eyu Gong Yi for CS2103. It is a Chat bot program.
````
Hi Gong Yi, love your chatbot, hehe looks cute!

martout2002 commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+public class Commands {
+    public enum Types {
```
wow wow wow, i like that you enum your types!

martout2002 reviewed (commented):
LGTM you seem to understand OOP concepts well!

lancehaha commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+    public static void runCommand(Types t, String input, TaskList taskList) {
+        try {
+            if (input == null) {
+                switch (t) {
```
for our module's special style norm indentation for switch and case is not recommended

lancehaha commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+                        break;
+                }
+            } else {
+                switch (t) {
```
for our module's special style norm indentation for switch and case is not recommended

lancehaha commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+            if (input == null) {
+                switch (t) {
+                    case LIST:
+                        list(taskList);
```
for our module's special style norm indentation for switch and case is not recommended

zaidansani replied:
seconding this! just wanted to add a quick tip shared on the forums, incase you have a lot of switch cases all around that you want to change - [configuring code style](https://se-education.org/guides/tutorials/intellijCodeStyle.html)

lancehaha commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+            } else {
+                switch (t) {
+                    case TODO:
+                        todo(taskList, input);
```
for our module's special style norm indentation for switch and case is not recommended

zaidansani commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+
+    }
+
+    //Command Logic
```
the header comments can be more elaborate and follow the javadocs standard, which can help other people in understanding how the method works without delving into the implementation & show up as a useful tooltip in your IDE!

zaidansani commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,39 @@
+        return this.tasks.get(i);
+    }
+    public void mark(int i) {
+        this.tasks.get(i).mark();
```
based on the coding standard, it is recommended to minimise the use of `this`. for example, in this, since there's no other variable named `tasks` which can cause confusion (for example in the constructor when you pass in values) - there is no need for `this.tasks...` here.

i believe it's the same for the other classes as well.

![image](https://github.com/user-attachments/assets/d50275ab-a4a5-499b-bf14-977807fbd739)

^ attached code standard screenshot for ease of reference

zaidansani commented on src/main/java/Commands.java:
```diff
@@ -0,0 +1,155 @@
+            int i = Integer.parseInt(input);
+            System.out.printf("Task %d: [%s] has been marked", i, taskList.get(i - 1));
+            taskList.mark( i- 1);
+        } catch (NumberFormatException e) {
```
generally, well done in ensuring the K&R style brackets are kept everywhere!

zaidansani commented on src/main/java/Diomon.java:
```diff
@@ -0,0 +1,24 @@
+        Scanner scanner = new Scanner(System.in);
+        TaskList taskList = new TaskList();
+        greeting();
+        while(true) {
```
it would be better to have a whitespace here: e.g `while (true)`

zaidansani reviewed (commented):
hi gong yi, i generally left comments on improvement - that even bothered on slightly nitpicky, but generally your coding standard follows the described convention VERY well - it would be way too much for me to leave positive comments everywhere.

adding javadocs would go a long way in making your code even more readable! good job!

EyuGongYi committed:
Add save feature and refactor code

EyuGongYi committed:
Add Load data feature, Parser class and refactor code

EyuGongYi committed:
Merge branch 'branch-Level-7'

EyuGongYi committed:
Add Date and Time

EyuGongYi committed:
Merge branch 'branch-Level-8'

EyuGongYi committed:
Add Package

EyuGongYi committed:
Merge branch 'add-gradle-support'

EyuGongYi committed:
Add and configure Gradle

EyuGongYi committed:
Add JUnit Tests

EyuGongYi committed:
Add JavaDoc

EyuGongYi committed:
Fix Coding standard

EyuGongYi committed:
Fix Coding Standard

EyuGongYi committed:
Add Find feature

EyuGongYi committed:
Merge branch 'branch-A-JavaDoc'

EyuGongYi committed:
Merge branch 'branch-CodingStandard'

# Conflicts (Resolved):
#	src/main/java/Main.java
#	src/main/java/diomon/Commands.java

EyuGongYi committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Diomon/TaskList.java
#	src/main/java/diomon/Commands.java

EyuGongYi committed:
Fix Find feature

EyuGongYi committed:
Add JavaFx

EyuGongYi committed:
Fix bugs

EyuGongYi committed:
Change dialog looks and message

EyuGongYi committed:
Merge branch 'branch-Level-10'

EyuGongYi committed:
Add Assertion

The program has user input that can mess up assumptions made
when creating the program.

Place assertions checks in the command class

Command Class and its functionality has the closest contact with the user input
 and it cant help prevent the most amount of error down the line

EyuGongYi committed:
Merge pull request from EyuGongYi/branch-A-Assertions

Add Assertion

EyuGongYi committed:
Improve code quality

Fix import statement
Remove unused code
Rename variables with name temp (temporary) to something else more
specific.

EyuGongYi committed:
Merge remote-tracking branch 'origin/master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/diomon/command/AddDeadlineCommand.java (Resolved)
#	src/main/java/diomon/command/AddEventCommand.java (Resolved)
#	src/main/java/diomon/command/AddTodoCommand.java (Resolved)
#	src/main/java/diomon/command/FindCommand.java (Resolved)
#	src/main/java/diomon/command/MarkCommand.java (Resolved)
#	src/main/java/diomon/command/UnmarkCommand.java (Resolved)

EyuGongYi committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/diomon/command/AddDeadlineCommand.java (Resolved)
#	src/main/java/diomon/command/AddEventCommand.java (Resolved)
#	src/main/java/diomon/command/AddTodoCommand.java (Resolved)
#	src/main/java/diomon/command/FindCommand.java (Resolved)
#	src/main/java/diomon/command/MarkCommand.java (Resolved)
#	src/main/java/diomon/command/UnmarkCommand.java (Resolved)

EyuGongYi committed:
Merge and fix branch 'origin/branch-A-CodeQuality'

#Conflicts:
#	src/main/java/diomon/command/AddDeadlineCommand.java

EyuGongYi committed:
Merge pull request #3 from EyuGongYi/branch-A-CodeQuality

Branch a code quality

EyuGongYi committed:
Fix code quality

Previous code quality changes got undone

Unused import statement in AddDeadlineCommand class has been removed

EyuGongYi committed:
Merge branch 'master' of https://github.com/EyuGongYi/ip

EyuGongYi committed:
Add help feature, Mass deleting, marking of tasks

EyuGongYi committed:
Merge branch 'branch-BCD-Extension'

EyuGongYi committed:
Add User guide

EyuGongYi committed:
Add user guide

EyuGongYi committed:
Edit User guide

EyuGongYi committed:
More edits to user guide

EyuGongYi committed:
Update Junit test

JUnit test was not updated to reflect the new refactored code

EyuGongYi committed:
Add Assertion, fix bug

Ensures that Function like delete, mark unmark actually did the change,
instead of giving false positive
Find function not displaying its response correctly

Asserts the state of the task list and task at the end of the operation.
Update the find response

EyuGongYi committed:
Fix functionally of bye

Does not show good bye message before exiting, can seem abrupt to user

Add a pause and shows the message  before exiting
